File: hw/Pérez/U3/HW23MazeGenerator/MazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Room.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public abstract class Room {
    private List<Door> doors;
    private Wall[] walls; 

    public Room() {
        doors = new ArrayList<>();
        walls = new Wall[4]; 
        for (int i = 0; i < 4; i++) {
            walls[i] = new Wall();
        }
    }

    public void addDoor(Door door, int wallIndex) {
        doors.add(door);
        if (wallIndex >= 0 && wallIndex < walls.length) {
            walls[wallIndex].setDoor(true); 
        }
    }

    public List<Door> getDoors() {
        return doors;
    }

    public Wall[] getWalls() {
        return walls;
    }

    public int getNumberOfDoors() {
        return doors.size();
    }

    public abstract String getRoomType();
}

File: hw/Pérez/U3/HW23MazeGenerator/MazeGeneratorJS/model/Room.js
import { Door } from './Door.js';
import { Wall } from './Wall.js';

export class Room {
    constructor() {
        this.doors = [];
        this.walls = Array.from({ length: 4 }, () => new Wall());
    }

    addDoor(door, wallIndex) {
        this.doors.push(door);
        if (wallIndex >= 0 && wallIndex < this.walls.length) {
            this.walls[wallIndex].setDoor(true);
        }
    }

    getDoors() {
        return this.doors;
    }

    getWalls() {
        return this.walls;
    }

    getNumberOfDoors() {
        return this.doors.length;
    }

    getRoomType() {
        throw new Error("This method should be overridden by subclasses");
    }
}
